File: Algorithms/Sorting/C#/BubbleSort.cs
using System;

public class Sorting
{
    public static void Main(string[] args)
    {

        // Non-Generic Bubble Sort Implementation for primitive int array
        void BubbleSort(int[] array) 
        {
            for (int i = 0; i < array.Length; i++)
            {
                for (int j = 0; j < array.Length - i - 1; j++)
                {
                    if (array[j].CompareTo(array[j + 1]) > 0)
                    {
                        var temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                    }
                }
            }
        }


        // Generic Bubble Sort Implementation for any Comparable Type
        void GenericBubbleSort<T>(T[] array) where T : IComparable  
        {
            for (int i = 0; i < array.Length; i++)
            {
                for (int j = 0; j < array.Length - i - 1; j++)
                {
                    if (array[j].CompareTo(array[j + 1]) > 0)
                    {
                        var temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                    }
                }       
            }
        }


        // Sorting primitive int array
        int[] numbers = { 8, 20, 3, 43, 12, 125, 86, 63, 28 };
        BubbleSort(numbers);

        foreach (int sortedNumbers in numbers)
        {
            Console.Write(sortedNumbers + " ");
        }
        Console.WriteLine();


        // Sorting String array (Generic version)
        string[] names = { "Oliver", "Nathan", "Ellie", "Jorge", "Eleni" };
        GenericBubbleSort(names);

        foreach (string sortedNames in names)
        {
            Console.Write(sortedNames + " ");
        }
        Console.WriteLine();


        // Sorting Integer array (Generic version). 
        // Although it has already been sorted in the non-generic version, this demonstrates the usage of the generic sorting method.
        GenericBubbleSort(numbers);

        foreach (int sortedNumbers in numbers)
        {
            Console.Write(sortedNumbers + " ");
        }
    }
}

File: Algorithms/Sorting/Python/BubbleSort.py
def bubble_sort(arr):
    for i in range(0, len(arr)):
        for j in range(0, len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
                #arr[j], arr[j + 1] = arr[j + 1], arr[j]


# Sorting integers
numbers = [8, 20, 3, 43, 12, 125, 86, 63, 28]
bubble_sort(numbers)
for sortedNumbers in numbers:
    print(sortedNumbers, end=" ")

print()
# Sorting strings
names = ["Oliver", "Nathan", "Ellie", "Jorge", "Eleni"]
bubble_sort(names)
for sortedNames in names:
    print(sortedNames, end=" ")

# Note: In contrast to languages like Java and C#, Python's dynamic typing and duck typing enable this function to work with
# various data types, similar to how a generic function behaves in Java or C#.
